javascript/ql/lib/semmle/javascript/dataflow/TaintTracking.qll

```diff
@@ -1126,6 +1126,21 @@ module TaintTracking {
     )
   }
 
+  /**
+   * Holds if `test` is of the form `typeof x === "something"`, preventing `x` from being a string in some cases.
```

Comment:
> ... in some cases

Are you hedging against the special case of `typeof new String("foo")`?

Reply:
I just copy-pasted the existing QLDoc from the old sanitizer-guard that the implementation came from.  

I think the "in some cases" can just be removed.

Reply:
I think "in some cases" referred to the fact that it depends on the value of `"something"` and the branch taken.